packages/components/src/popover/index.js

```diff
@@ -250,7 +252,39 @@ const Popover = (
 		__unstableShift
 			? shift( {
 					crossAxis: true,
-					limiter: limitShift(),
+					limiter: limitShift( {
+						offset: ( { placement: currentPlacement } ) => {
+							// The following calculations are aimed at allowing the floating
+							// element to shift fully below the reference element, when the
+							// reference element is in a different document (i.e. an iFrame).
+							if ( ownerDocument === document ) {
+								return 0;
+							}
+
+							// The main axis (according to floating UI's docs) is the "x" axis
+							// for 'top' and 'bottom' placements, and the "y" axis for 'left'
+							// and 'right' placements.
+							const mainAxis = isTopBottomPlacement(
```

Comment:
I guess we can extract a util to just return the `mainAxis and crossAxis`..

Reply:
I'm not sure if would be reducing the amount of code by much — this is because the `shift` and the `offset` middlewared use opposite logic to identify the main and cross axis. For example, if the `placement` is `top`, the main axis for the `offset` middleware is the `y` axis, while the main axis for the `shift` middleware is the `x` axis. This is just how floating ui works 🤷 

Do you still think it would be useful to abstract this logic in a separate function?

Reply:
Oh sorry, I missed that.. It's good as is.